codeql.yml

```diff
@@ -0,0 +1,100 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# For more information see: 
+#     https://nasa.github.io/scrub/
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [main, develop]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [develop]
```

Comment:
This probably isn't necessary since this was already checked when pushed to branch `develop`? But not a huge deal.

Reply:
I was going for the auto-checks at PR time before it hits the branch, but yeah when it's merged/pushed it may run again. However, there could also be other changes by that time. 
<img width="825" alt="Screen Shot 2024-03-14 at 7 26 48 PM" src="https://github.com/NASA-AMMOS/slim-starterkit-python/assets/3822522/85d17d55-e26b-4585-af14-8c96a536105a">